packages/docusaurus-theme-bootstrap/src/theme/DocSidebar/index.tsx

```diff
@@ -11,57 +11,63 @@ import isInternalUrl from '@docusaurus/isInternalUrl';
 import {NavItem, Nav, Button} from 'reactstrap';
 import useLockBodyScroll from '@theme/hooks/useLockBodyScroll';
 import classNames from 'classnames';
+import {Props} from '@theme/DocSidebar';
+import type {PropSidebarItem} from '@docusaurus/plugin-content-docs-types';
 
 import styles from './styles.module.css';
 
-const DocSidebarItem = ({item, onItemClick, ...props}) => {
-  const {items, href, label, type} = item;
-
-  switch (type) {
-    case 'category':
-      return (
-        items.length > 0 && (
-          <div>
-            <h4 className="ml-2">{label}</h4>
-            {items.map((childItem) => (
-              <DocSidebarItem
-                key={childItem.label}
-                item={childItem}
-                onItemClick={onItemClick}
-              />
-            ))}
-          </div>
-        )
-      );
+type DocSidebarItemProps = {
+  item: PropSidebarItem;
+  onItemClick: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
+};
 
-    case 'link':
-    default:
-      return (
-        <NavItem>
-          <Link
-            key={label}
-            className="sidebar-item m-4 text-white"
-            to={href}
-            {...(isInternalUrl(href)
-              ? {
-                  isNavLink: true,
-                  activeClassName: 'active',
-                  exact: true,
-                  onClick: onItemClick,
-                }
-              : {
-                  target: '_blank',
-                  rel: 'noreferrer noopener',
-                })}
-            {...props}>
-            {label}
-          </Link>
-        </NavItem>
-      );
+const DocSidebarItem = ({
+  item,
+  onItemClick,
+}: DocSidebarItemProps): JSX.Element => {
+  if (item.type === 'category') {
```

Comment:
I changed the switch to an if, because typescript can understand what type of the union it is. 
`PropSidebarItem = PropSidebarItemLink | PropSidebarItemCategory`
using if typescript understand which specific type is being used, and allows the access to adequate properties

Reply:
sure :) didn't know TS was not able to handle discriminated union types with switch